### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `oklch` and `oklab`

### ISSUE THREAD

**Issue by ghost** [stylelint/stylelint#7241]

**Comment by Mouvedia** [comment]
We are waiting for the next release of csstree.
It's fixed on HEAD.

ref: csstree/csstree#244

**Comment by jeddy3** [comment]
@meerhimmel While waiting on the next release of csstree (the upstream package that powers this rule), you can amend the syntax in your own config:

```json
{
  "rules": {
    "declaration-property-value-no-unknown": [true, {
      "typesSyntax": { 
        "color": "| oklch( [ <percentage> | <number> | none] [ <percentage> | <number> | none] [ <hue> | none] [ / [<alpha-value> | none] ]? )"
      }
    }]
  }
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYLIBsICd4ogGtMwALACgBYAGAOgEYBSFO+gJgGYU37aBWAOwBKANwoAviAA0IAGYBLTDABySALZxEMAB7qADktpgAzsengIAOwUBzBCGAAdSyhSOQOAK5Lj7-E5dXNxBYMEwkHCQAF3krAFo9HAg9GBwogE84gDckTE8YOMsIOM9LQiKAd0s-FABtKK8YKTRnIKD3DJTjAGV0yyikbRr0Vrb2i2wcGvcAHwJiMnI6lAAeFJwwGH6kGxgAPhQ5lctPNQAjVIO5ossYAF1ltdTN7d2r1ZPzy8OUG-vH0j5d5-B61FAAejqK1yelISGyuSBPxBKDuAH4UEJ3KM2uIcRI7qM8ZZJDJINZ5DYAGK4NTRewAK2MVnMsD0ZkQAVc7mMGSUmHk-WmIHCURgvPcUlGPL5MAF-Ti5NscV5SEsUAiUGFovFUWxJJA4iAA).

**Comment by Mouvedia** [comment]
I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by Mouvedia** [comment]
> Unexpected unknown value \"c\" for property \"background-color\" (declaration-property-value-no-unknown)

The challenge for this one is to add support for the channel keywords of `lab` and `lch`.
i.e. `l`, `a`, `b`, `c`, `h`
ref https://drafts.csswg.org/css-color-5/#channel-keyword

**Comment by romainmenke** [comment]
> The challenge for this one is to add support for the channel keywords of lab and lch.
i.e. l, a, b, c, h and alpha

This is bugged and incomplete, but this snippet illustrates an approach to adding support for these keywords.

Instead of defining them as valid syntax we could also eagerly replace them by numbers nodes.

Ideally we would only replace them in valid positions and only specific keywords for specific functions.


```js
function patch_relative_color_keywords(csstree_value_node) {
	let color_function_stack = 0;
	let relative_color_functions = /^(?:rgb|rgba|hsl|hsla|hwb|lab|lch|oklab|oklch|color)$/i;
	let relative_color_channel_keywords = /^(?:r|g|b|h|s|l|a|b|c|w|b|x|y|z)$/i;

	walk(
		csstree_value_node,
		{
			enter: function (node) {
				if (
					node.type === 'Function' &&
					relative_color_functions.test(node.name)
				) {
					color_function_stack++;

					return;
				}

				if (
					node.type === 'Identifier' &&
					color_function_stack > 0 &&
					relative_color_channel_keywords.test(node.name)
				) {
					node.type = 'Number';
					node.value = '0';
					delete node.name;

					return;
				}
			},
			exit: function (node) {
				if (
					node.type === 'Function' &&
					relative_color_functions.test(node.name)
				) {
					color_function_stack--;

					return;
				}
			},
		},
	);
}
```

**Comment by Mouvedia** [comment]
As long as we have good JSDoc description for those hacky fixes, I would be fine with it.
I would prefer if it was supported natively by `css-tree` obviously.

> [Will publish a new major version this week.](https://github.com/csstree/csstree/issues/244#issuecomment-2329225772)

You might be able to sneak a PR before then.

**Comment by romainmenke** [comment]
This was also fixed in https://github.com/stylelint/stylelint/pull/8850

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvChANZ5gAWAFACwAMAdAIwCkKzLATAMwrcWTAKwB2AJQBuLDnxES5SrQBmhCAFsUhGFBR5cKKigD0KJHgAOVJFKwBfEABoQygJZ4YAOSTq4iGAAePhYeTGAAzuFO4BCYbgDmCCAY2CjoIIQArh7h6SQpODjpsGB4SIRIAC6usQC0FmoWMISVAJ61AG7mmTC1mBC1mZhk-QDumHkoANqVWTAAujIodvbRkHGu8QBiROpVSUzhbR54rpizYEwAVuGx0bAWUYgFaSBHrSdnlZPpZZUwR3SjiW6Xen3OtXWCVqRyQmCg5SgPxAfwB3xAqzsQA)
